lib/system/iterators_1.nim

```diff
@@ -23,22 +23,24 @@ iterator countdown*[T](a, b: T, step: Positive = 1): T {.inline.} =
       for i in countdown(9, 2, 3):
         i
     assert y == @[9, 6, 3]
-  when T is (uint|uint64):
-    var res = a
-    while res >= b:
-      yield res
-      if res == b: break
-      dec(res, step)
-  elif T is IntLikeForCount and T is Ordinal:
-    var res = int(a)
-    while res >= int(b):
-      yield T(res)
-      dec(res, step)
-  else:
-    var res = a
-    while res >= b:
-      yield res
-      dec(res, step)
+  try:
```

Comment:
Hm, I'm not so sure about this. `try`/`except` has a certain amount of overhead. Why can't the exceptional conditions be checked for?

Reply:
The exceptional condition is specified as part of the `inc`/`dec` API: "If such a value does not exist, `OverflowDefect` is raised or a compile time error occurs." That this code worked in the first place is because those functions *don't* work as specced for builtin types!

https://nim-lang.org/docs/system.html#inc%2CT%2Cint
https://nim-lang.org/docs/system.html#dec%2CT%2Cint